math/src/main/java/com/powsybl/math/matrix/DenseMatrix.java

```diff
@@ -233,12 +237,54 @@ public LUDecomposition decomposeLU() {
 
     @Override
     public Matrix times(Matrix other) {
-        return new DenseMatrix(toJamaMatrix().times(other.toDense().toJamaMatrix()));
+        return times(other.toDense());
+    }
+
+    public DenseMatrix times(DenseMatrix other) {
+        Objects.requireNonNull(other);
+        if (other.rowCount != columnCount) {
+            throw new MatrixException("Invalid matrices inner dimension");
+        }
+
+        DenseMatrix result = new DenseMatrix(rowCount, other.columnCount);
+
+        double[] otherColumnJ = new double[columnCount];
+        for (int j = 0; j < other.columnCount; j++) {
+            for (int k = 0; k < columnCount; k++) {
+                otherColumnJ[k] = other.getUnsafe(k, j);
+            }
+            for (int i = 0; i < rowCount; i++) {
+                double s = 0;
+                for (int k = 0; k < columnCount; k++) {
+                    s += getUnsafe(i, k) * otherColumnJ[k];
+                }
+                result.setUnsafe(i, j, s);
+            }
+        }
+
+        return result;
+    }
+
```

Comment:
In the same way as add(), wouldn't it be interesting to have more general time() function such as times(DenseMatrix other, Double scalar) to speed up operations like [A] * [B] * scalar ?
Would it be relevant to have a time(Double scalar) function as well?

Reply:
[A] * [B] * scalar added